File: fintechplatform/src/main/java/com/fintechplatform/android/secure3d/ui/Secure3DPresenterModule.java
package com.fintechplatform.android.secure3d.ui;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ingrid on 12/12/17.
 */
@Module
public class Secure3DPresenterModule {

    private final Secure3DContract.View view;

    public Secure3DPresenterModule(Secure3DContract.View view) {
        this.view = view;
    }

    @Singleton
    @Provides
    Secure3DContract.Presenter providesSecure3DPresenter(){
        return new Secure3DPresenter(view);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/secure3d/di/Secure3DPresenterModule.kt
package com.fintechplatform.ui.secure3d.di

import com.fintechplatform.ui.secure3d.Secure3DContract
import com.fintechplatform.ui.secure3d.Secure3DPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
open class Secure3DPresenterModule(private val view: Secure3DContract.View) {

    @Singleton
    @Provides
    internal fun providesSecure3DPresenter(): Secure3DContract.Presenter = Secure3DPresenter(view)
    
}